### PR TITLE
Implemented charge balance timer.

### PR DESCRIPTION
**Pull request by wfox5812**
## Issue ticket number and issue description

Ticket #  -> Fix X

## Describe your changes

Changes made
-
-
-

### CODE REVIEW

**Comment by wfox5812** [comment]
Didn't declare the variable for the state of the charge balance timer, needs to be added once the state machine initialization is ready.

File: shepherd_bms/src/main.cpp [hunk 1]
```diff
@@ -72,11 +72,32 @@ void balanceCells(AccumulatorData_t *bms_data)
...
+
 	if (!compute.isCharging()) return false;
 	if (bmsdata->maxTemp.val > MAX_CELL_TEMP_BAL) return false;
 	if (bmsdata->maxVoltage.val <= (BAL_MIN_V * 10000)) return false;
 	if(bmsdata->deltVoltage <= (MAX_DELTA_V * 10000)) return false;
 
+	//balance timer state 0 == not balancing
+	//balance timer state 1 == balancing
+
+	if (balanceTimer.isTimerExpired() && balanceTimerState == 0)
```
Comment (nwdepatie) [inline shepherd_bms/src/main.cpp:85]: nit - either make it a boolean ( ```if (isBalancing)```  ) or enumerate these states ( ``` if (balanceTimerState == NOT_BALANCING)``` ) or something, try and avoid magic numbers where possible

**Review by nwdepatie — CHANGES_REQUESTED** [review]
So this PR is gonna stay in PR land for a hot sec until we get the state initialization functionality implemented, but overall I think it looks good, just a small nitpick. That being said, we do have to test this functionality before merging in on hardware

**Review by mccauleyma — CHANGES_REQUESTED** [review]
The other element to this ticket is once timings are implemented, we want to add to the openCellVoltage that it should not update when balancing and then make the balancing algorithm use open cell voltages, instead of live.

This will allow for a smarter balance as it will account for the internal resistance impact

**Comment by mccauleyma** [comment]
@dyldonahue I assume it is up to you if you want my request to be a part of this PR/ticket or if it needs its own ticket. But the whole point of adding the timers is to implement the better balancing I mention, so imo they are pretty linked in terms of features, but could be easily split in terms of code

**Comment by dyldonahue** [comment]
will weigh options, based on timeline may end up just being a quick one session fix that gets done asap so might not matter either way

**Comment by dyldonahue** [comment]
@wfox5812 forgot to mention last week but if u wanted to add onto this with additions @mccauleyma mentioned in the comment above go for it

**Comment by nwdepatie** [comment]
Closed as we made an updated PR https://github.com/Northeastern-Electric-Racing/shepherd_bms/pull/177

**Pull request closed without merging**